addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4105,8 +4106,12 @@ export class OdooEditor extends EventTarget {
         }
 
         // Clean custom selections
-        if (this.deselectTable() && hasValidSelection(this.editable)) {
-            this.document.getSelection().collapseToStart();
+        if (hasTableSelection(this.editable)) {
+            if (!(this.options.customGradientButtonEl && this.options.customGradientButtonEl.classList.contains('o_custom_gradient_btn'))) {
+                this.deselectTable();
+            }
+            const selection = this.document.getSelection();
+            selection.rangeCount && selection.collapseToStart();
         }
```

Comment:
@Zynton Following the recent changes, my runbot is now displaying a red status because it failing `post_install from auth_totp to microsoft_outlook` build The issue centers around the inability to locate the 'default view' within the 'cover' function as specified in the 'snippets.editor.js' file. Despite my efforts to investigate, it appears that the current conditions, as well as the previous conditions, are not being met. This raises the question of why the build is failing now, especially considering that the same fix resulted in a green status in the last successful build.

Reply:
It seems that `targetEl.ownerDocument` is undefined, which would indicate that `targetEl` is not in the DOM. Have you checked what that target is supposed to be? Any information you can gather as to why it's not in the DOM and whether it should be or if this function should instead not be called?

Reply:
@Zynton I investigated it and found that `targetEl` is in DOM and we have `targetEl.ownerDocument` but we don't have it's `defaultView` thats why we get this traceback.

Reply:
Now there's something I've never seen... :open_mouth: The only case I can think of where this can happen is if the `ownerDocument` is a `DocumentFragment`. Is it? If so, _how come_?

Reply:
@visp-odoo I took a look on Runbot and it seems that the issue is this is called after the editor gets destroyed. The view doesn't exist anymore. The name of the game now is to find our what in your changes causes that difference.

Reply:
@Zynton it seems that after investigating, I found that the flow remains the same both with and without my modifications. During debuging i found that view is not changed still it throws a traceback.As we can see in img below...
![image](https://github.com/odoo/odoo/assets/98956978/629dab05-cb35-46fb-9ab9-1f787da3e15d)

IMO, there is difference in process sequence or migth be some other thing.

Reply:
@Zynton Can we just put check for `targetEl.ownerDocument.defaultView` ? I'm stranded in this one.

Reply:
OK so I investigated and the cause seems to be the following chain of events.

Before your changes, `deselectTable` was always called at the end of `cleanForSave`, which in turn called `observerUnactive`, causing `observerFlush`, `observerApply` and finally the `content_changed` event. `Wysiwyg` listens to this event and responds by calling `updateCurrentSnippetEditorOverlay`, which calls `cover`. In consequence, this whole chain is triggered as a direct effect of `cleanForSave`, therefore before destroying the editor:
![image](https://github.com/odoo/odoo/assets/24205914/a14e8b1f-089a-4a52-831d-49561916a88e)
So `targetEl` is still in the DOM and all is well.

After your changes, in this case, `deselectTable` is not called at the end of `cleanForSave` and the first time we call `observerUnactive` and cause this whole chain is in the editor's `destroy` function, which is only called after the whole widget was removed from the DOM:
![image](https://github.com/odoo/odoo/assets/24205914/fe787269-7f39-49f3-80f7-5a4f8ad77a33)
So `targetEl` is not in the DOM anymore and all hell breaks loose.

Solution: call `observerFlush` as a very last step of the editor's `cleanForSave` function. @Goaman as observer expert, do you agree it makes sense to add that call there conceptually?

Reply:
@Zynton I'm ok with it